packages/utilities/src/dom.ts

```diff
@@ -204,6 +204,29 @@ export function getRect(element: HTMLElement | Window | null): IRectangle | unde
   return rect;
 }
 
+/**
+ * Determines if an element, or any of its ancestors, contian the given attribute
+ * @param element - element to start searching from
+ * @param attribute - the attribute to search for
+ * @returns the value of the first instance found
+ */
+export function elementContainsAttribute(element: HTMLElement, attribute: string): string | undefined {
```

Comment:
I feel like I've seen this code repeated elsewhere.

Could we just have one method to find a node which matches some criteria?

Maybe `findElementRecursive(cb: (el) => boolean): HTMLElement` or something:

```tsx
// hmm better name?
findElementRecursive(el => el.hasAttribute(attr));
```
Then you could reuse the "go up the parents" logic elsewhere.

getElementAttributeRecursive(el, attr) {
  const el = findElementRecursive(el => el.hasAttribute(attr));

  return el && el.getAttribute(attr);
}
```

Reply:
Yeah, I'll look into this